packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,257 @@
+    uint256 instanceId,
+    ClaimParameters memory claimParameters
+  ) external override adminRequired {
+    Claim memory claim = _getClaim(creatorContractAddress, instanceId);
+    if (claimParameters.endDate != 0 && claimParameters.startDate >= claimParameters.endDate)
+      revert IGachaLazyClaim.InvalidStartDate();
+    if (claimParameters.totalMax != claim.totalMax) revert IGachaLazyClaim.CannotChangeTotalMax();
+    if (claimParameters.startingTokenId != claim.startingTokenId) revert IGachaLazyClaim.CannotChangeStartingTokenId();
+    if (claimParameters.itemVariations != claim.itemVariations) revert IGachaLazyClaim.CannotChangeItemVariations();
+    if (claimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    if (keccak256(abi.encodePacked(claimParameters.location)) != keccak256(abi.encodePacked(claim.location))) {
+      revert IGachaLazyClaim.CannotChangeLocation();
+    }
+    if (claimParameters.paymentReceiver != claim.paymentReceiver) revert IGachaLazyClaim.CannotChangePaymentReceiver();
+    if (claimParameters.erc20 != claim.erc20) revert IGachaLazyClaim.CannotChangePaymentToken();
+
+    // Overwrite the existing values for startDate, endDate, and cost
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claim.storageProtocol,
+      total: claim.total,
+      totalMax: claim.totalMax,
+      startDate: claimParameters.startDate,
+      endDate: claimParameters.endDate,
+      startingTokenId: claim.startingTokenId,
+      itemVariations: claim.itemVariations,
+      location: claim.location,
+      paymentReceiver: claim.paymentReceiver,
+      erc20: claim.erc20,
+      cost: claimParameters.cost
+    });
+    emit GachaClaimUpdated(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaim}.
+   */
+  function getClaim(address creatorContractAddress, uint256 instanceId) public view override returns (Claim memory) {
+    return _getClaim(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaimForToken}.
+   */
+  function getClaimForToken(
+    address creatorContractAddress,
+    uint256 tokenId
+  ) external view override returns (uint256 instanceId, Claim memory claim) {
+    instanceId = _claimTokenIds[creatorContractAddress][tokenId];
+    claim = _getClaim(creatorContractAddress, instanceId);
+  }
+
+  function _getClaim(address creatorContractAddress, uint256 instanceId) private view returns (Claim storage claim) {
+    claim = _claims[creatorContractAddress][instanceId];
+    if (claim.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.ClaimNotInitialized();
+  }
+
+  /**
+   * See {IGachaLazyClaim-mintReserve}.
+   */
+  function mintReserve(MintReservation calldata mintReservation) external payable override {
+    // _validateEOA(); do we need a validate EOA or validateSigner?
+    _mintReserve(mintReservation);
+  }
+
```

Comment:
function mintReserve(address creatorContractAddress, uint256 instanceId, uint32 amount) external payable override {
   Claim storage claim = _getClaim(creatorContractAddress, instanceId);
   // CHECK msg.value == (claim.cost + MINT_FEE)*amount;
   uint32 amountAvailable = claim.totalMax - claim.total;
   if (amountAvailable == 0) revert SoldOut
   uint32 amountToBuy = min(amount, amountAvailable);
   claim.total += amountToBuy;
   // increment user counts by amountToBuy
   // forward value to payment receiver in the amount of amountToBuy*claim.cost
   if (amountToBuy < amount) {
    // refund (amount-amountToBuy)*(claim.cost + MINT_FEE) to sender
   }
}

Reply:
ended up modifying the CHECK msg.value == (claim.cost + MINT_FEE)amount; ->     if (msg.value < (claim.cost + MINT_FEE) * mintCount) revert IGachaLazyClaim.InvalidPayment();

as well as to match:
if (amountToBuy < amount) { -> if (msg.value > (claim.cost + MINT_FEE) * amountToReserve){